# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/ffvpx/libavcodec/idctdsp.h
# Commit: 186e2a334f2d
# Full Hash: 186e2a334f2deb37b29f44b28bc8c83babeeb9a4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-09 09:37:17
# Regressor Bug: 1877221
# File Overlap Count: 1
# Description:
#   Bug 1877221 - Pull in new ffmpeg code, reapply patch. r=media-playback-reviewers,chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199909
# ==============================================================================

diff -r c5920de78d16 -r 186e2a334f2d media/ffvpx/libavcodec/idctdsp.h
--- a/media/ffvpx/libavcodec/idctdsp.h	Thu Feb 08 10:39:53 2024 +0000
+++ b/media/ffvpx/libavcodec/idctdsp.h	Thu Feb 08 10:39:54 2024 +0000
@@ -19,11 +19,12 @@
 #ifndef AVCODEC_IDCTDSP_H
 #define AVCODEC_IDCTDSP_H
 
+#include <stddef.h>
 #include <stdint.h>
 
 #include "config.h"
 
-#include "avcodec.h"
+struct AVCodecContext;
 
 enum idct_permutation_type {
     FF_IDCT_PERM_NONE,
@@ -95,23 +96,23 @@
 void ff_add_pixels_clamped_c(const int16_t *block, uint8_t *av_restrict pixels,
                              ptrdiff_t line_size);
 
-void ff_idctdsp_init(IDCTDSPContext *c, AVCodecContext *avctx);
+void ff_idctdsp_init(IDCTDSPContext *c, struct AVCodecContext *avctx);
 
-void ff_idctdsp_init_aarch64(IDCTDSPContext *c, AVCodecContext *avctx,
+void ff_idctdsp_init_aarch64(IDCTDSPContext *c, struct AVCodecContext *avctx,
                              unsigned high_bit_depth);
-void ff_idctdsp_init_alpha(IDCTDSPContext *c, AVCodecContext *avctx,
+void ff_idctdsp_init_alpha(IDCTDSPContext *c, struct AVCodecContext *avctx,
                            unsigned high_bit_depth);
-void ff_idctdsp_init_arm(IDCTDSPContext *c, AVCodecContext *avctx,
+void ff_idctdsp_init_arm(IDCTDSPContext *c, struct AVCodecContext *avctx,
                          unsigned high_bit_depth);
-void ff_idctdsp_init_ppc(IDCTDSPContext *c, AVCodecContext *avctx,
+void ff_idctdsp_init_ppc(IDCTDSPContext *c, struct AVCodecContext *avctx,
                          unsigned high_bit_depth);
-void ff_idctdsp_init_riscv(IDCTDSPContext *c, AVCodecContext *avctx,
+void ff_idctdsp_init_riscv(IDCTDSPContext *c, struct AVCodecContext *avctx,
                            unsigned high_bit_depth);
-void ff_idctdsp_init_x86(IDCTDSPContext *c, AVCodecContext *avctx,
+void ff_idctdsp_init_x86(IDCTDSPContext *c, struct AVCodecContext *avctx,
                          unsigned high_bit_depth);
-void ff_idctdsp_init_mips(IDCTDSPContext *c, AVCodecContext *avctx,
+void ff_idctdsp_init_mips(IDCTDSPContext *c, struct AVCodecContext *avctx,
                           unsigned high_bit_depth);
-void ff_idctdsp_init_loongarch(IDCTDSPContext *c, AVCodecContext *avctx,
+void ff_idctdsp_init_loongarch(IDCTDSPContext *c, struct AVCodecContext *avctx,
                                unsigned high_bit_depth);
 
 #endif /* AVCODEC_IDCTDSP_H */